editor/src/clj/editor/scene.clj

```diff
@@ -629,8 +645,9 @@
                                          (if (seq selected-aabbs)
                                            (reduce geom/aabb-union geom/null-aabb selected-aabbs)
                                            scene-aabb))))
-  (output scene-aabb AABB :cached (g/fnk [scene]
+  (output scene-aabb AABB :cached (g/fnk [scene active-view]
                                     (let [scene-aabb (calculate-scene-aabb geom/null-aabb geom/Identity4d scene)]
+                                      (reframe scene scene-aabb active-view)
```

Comment:
This is very very bad because the only job of the outputs in the graph is producing data, not performing side effects, especially not mutating the graph. In this case, `reframe` calls `set-camera!` which, in turn, changes the graph with `g/transact`. The act of reading changes what is being read. This may cause infinite loops and stuff.

Instead, any mutation of the graph should happen outside of the graph. After a quick look, I think that possible candidates for such reframing could be `refresh-scene-view!` or `update-image-view!` functions

Reply:
Regarding subscribing to changes of a value in a graph, I think this is a very useful feature, and we will definitely need something like that in the future. Maybe with an option to subscribe to multiple outputs in the same subscription. I have a [dev-time implementation](https://github.com/defold/defold/blob/dev/editor/src/reveal/editor/reveal.clj#L167-L184) of such an observable that I sometimes use for debugging. It's useful, and we currently re-implement it ad-hoc in various places in the editor UI.

Reply:
Fixed! Thank you!